core/drivers/imx_i2c.c

```diff
@@ -292,6 +327,9 @@ TEE_Result imx_i2c_read(uint8_t bid, uint8_t chip, uint8_t *buf, int len)
 	if ((len && !buf) || chip > 0x7F)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	if (!i2c_bus[bid].va)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
I think  ```TEE_ERROR_BAD_STATE``` is more adequate since this error is not related to parameters.

Reply:
I'd rather keep bad parameters: the way I see it, the user requested a read transaction for a bus that it didnt initialize